downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,759 @@
+```
+
+- version suggestions with the list of source code changes
+```
+$bal semver --show-diff
+checking for the latest compatible release version available in central...
+
+=========================================================================
+ Comparing version '1.2.2-SNAPSHOT'(local) with version '1.2.1'(central) 
+=========================================================================
+[+-] package 'io' is modified [version impact: MAJOR]
+  [+-] module 'io' is modified [version impact: MAJOR]
+    [++] function 'printlnNew' is added [version impact: MINOR]
+    [+-] function 'println' is modified [version impact: MAJOR]
+      [+-] documentation is modified [version impact: PATCH]
+      [--] 'isolated' qualifier is removed [version impact: AMBIGUOUS]
+      [++] 'transactional' qualifier is added [version impact: AMBIGUOUS]
+      [++] new required parameter 'a' is added [version impact: MAJOR]
+      [++] new defaultable parameter 'b' is added [version impact: MINOR]
+      [+-] parameter type changed from 'Printable' to 'string' [version impact: AMBIGUOUS]
+      [+-] function body is modified [version impact: PATCH]
+
+current version: 1.2.2-SNAPSHOT
+compatibility impact (compared with the release version '1.2.1'): backward-incompatible changes detected
+suggested version: 2.0.0
+```
+##### CLI
+
+Introduced the `bal graph` CLI command, which resolves the dependencies of the current package and prints the dependency graph in the console. This produces the textual representation of the dependency graph using the DOT graph description language.
+
+```ballerina
+$ bal graph
+digraph "org/package:0.1.0" {
+        node [shape=record]
+        "org/package" [label="<0.1.0> org/package:0.1.0"];
+        "ballerina/io" [label="<1.2.2> ballerina/io:1.2.2"];
+
+        // Edges
+        "org/package":"0.1.0" -> "ballerina/io":"1.2.2";
+}
+```
+
+##### Compiler API
+
+- Introduced a set of builders in the Types API, which are used to construct complex types that have varying components
+
+##### Ballerina Shell
+
+##### Ballerina Update Tool
+
+#### Improvements
+
+##### Language Server
+- Improved the `Create variable` code action in the `async send` action
+- Added completion support for the resource access action context
+
+##### OpenAPI Tool
+Added support for OpenAPI schema constraint properties in client/service generation. With this improvement, the OpenAPI constraints will be applied as `ballerina/constraint` standard library package annotations when generating Ballerina clients and services from the OpenAPI definition.
+The following OpenAPI properties are currently supported in the Ballerina OpenAPI generation tool.
+- `minimum`, `maximum`, `exclusiveMinimum`, and `exclusiveMaximum` for `integer` and `number` types
+- `minLength` and `maxLength` for `string` type
+- `minItems` and `maxItems` for `array` type
+
+To view bug fixes, see the GitHub milestone for 2201.2.0 (Swan Lake) of the repositories below.
```

Comment:
`#### Bug fixes` is missing

Reply:
Seems both urls are not having any issues. So can we remove the links @mohanvive / @keizer619 ?

Reply:
Removed.

Reply:
@praneesha It seems above url for Language Server is wrong. Below is the correct one, please add it, https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+label%3ATeam%2FLanguageServer+milestone%3A2201.2.0